test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1221,6 +1222,29 @@ public static <T> T randomValueOtherThanMany(Predicate<T> input, Supplier<T> ran
         return randomValue;
     }
 
+    /**
+     * Mutates a {@link Map} by either adding, updating, or removing an entry.
+     */
+    public static <K, V> Map<K, V> mutateMap(Map<K, V> original, Supplier<K> randomKeySupplier, Supplier<V> randomValueSupplier) {
+        Map<K, V> mapCopy = new HashMap<>(original);
+        if (original.isEmpty()) {
+            mapCopy.put(randomKeySupplier.get(), randomValueSupplier.get());
+        } else {
+            switch (randomIntBetween(1, 3)) {
+                case 1 -> mapCopy.put(randomKeySupplier.get(), randomValueSupplier.get());
+                case 2 -> {
+                    K someKey = randomFrom(original.keySet());
+                    mapCopy.put(someKey, randomValueOtherThan(original.get(someKey), randomValueSupplier));
+                }
+                case 3 -> {
+                    mapCopy.remove(randomFrom(mapCopy.keySet()));
+                }
+                default -> throw new IllegalStateException();
+            }
+        }
+        return mapCopy;
+    }
+
```

Comment:
I understand the idea, and I do think this is a useful method. However, I want us to be very sure before adding stuff to `ESTestCase` because it's used to heavily and it's quire large. 

What I am thinking is that we have a lot of random helpers but we do not have any other mutators. If we want to do that I would prefer to keep it in a separate PR. Maybe for now this can be a local method since you use it only in one file. 

What do you think?

Reply:
Maybe we can have a localised class near the tests, is that possible?

Reply:
I refactored something which resulted in me only using this method only in 1 class, so I've moved it to a method inside the `HealthInfoTests` class.